gamemodes/terrortown/entities/weapons/weapon_tttbasegrenade.lua

```diff
@@ -290,18 +456,199 @@ function SWEP:Initialize()
+			or (target.preventDamageChainReaction
+			and (hit_by_grenade or hit_by_grenade_projectile)
+			and inflictor:GetShotToActivate())
+		then
+			return
+		end
+
+
+		if not target:GetShotToActivate() then
+			target:SetShotToActivate(true)
+			if is_grenade and target.damageTriggerBehavior == 1 then
+				-- look for a spot above the ground
+				local pos = target:GetPos()
+				local tr = util.TraceLine({
+					start = pos,
+					endpos = pos + Vector(0,0,-32),
+					mask = MASK_SHOT_HULL,
+					filter = target,
+				})
+				-- pull out of the surface
+				local size = math.abs(target:OBBMaxs().z) + math.abs(target:OBBMins().z)
+				if tr.Hit then
+					pos = tr.HitPos + tr.HitNormal * 0.6
+					pos = pos + Vector(0, 0, size + 8)
+				end
+
+				-- make the grenade
+				local grenade = target:TryCreate(dmginfo:GetAttacker())
+				grenade:SetPos(pos)
+				grenade:EmitSound( grenade.pinSound )
+
+				-- try to inherit forces
+				local phys = grenade:GetPhysicsObject()
+				if IsValid(phys) then
+					local desiredVelocity = dmginfo:GetDamageForce()
+					local targetPhys = target:GetPhysicsObject()
+					if IsValid(targetPhys) then
+						desiredVelocity = desiredVelocity + targetPhys:GetVelocity()
+					end
+					phys:ApplyForceOffset( desiredVelocity, dmginfo:GetDamagePosition() )
+				end
+			elseif (is_grenade and target.damageTriggerBehavior == 2)
+				or (is_grenade_projectile and target.damageTriggerProjectileExplosion)
+			then
+				if is_grenade then
+					target:EmitSound( target.pinSound )
+				end
+				target:TryExplode(dmginfo:GetAttacker())
+			end
+		end
+	end)
+end
+
+if SERVER then
+	util.AddNetworkString("ttt2_quick_grenade")
+
+	local last_weapon = nil
+	net.Receive("ttt2_quick_grenade", function(_, ply)
+		if not IsValid(ply) or not ply:IsTerror() or not ply:Alive() then return end
+
+		local begin = net.ReadBool()
+
+		if begin then
+			local inventory = ply:GetInventory()
+			last_weapon = ply:GetActiveWeapon()
+			for _, iwep in pairs(inventory[WEAPON_NADE]) do
+				if IsValid(iwep) then
+					ply:SetActiveWeapon(iwep)
+					ply:ConCommand("+attack")
+					break
+				end
+			end
+		else
+			ply:ConCommand("-attack")
+			if IsValid(last_weapon) then
+				local vm = ply:GetViewModel()
+				local seq_id = vm:GetSequence()
+				local seq_info = vm:GetSequenceInfo(seq_id)
+				local anim_duration = seq_info.fadeouttime * ply:GetPlaybackRate()
+				timer.Simple(anim_duration, function()
+					if IsValid(last_weapon) then
+						ply:SelectWeapon(WEPS.GetClass(last_weapon))
+					else
+						ply:ConCommand("lastinv")
+					end
+				end)
+			else
+				ply:ConCommand("lastinv")
+			end
+		end
+	end)
+end
+
+if CLIENT then
+	local function QuickGrenade(begin)
+		net.Start("ttt2_quick_grenade")
+		net.WriteBool(begin)
+		net.SendToServer()
+	end
+
+	concommand.Add("+ttt2_quick_grenade", function(ply)
+		QuickGrenade(true)
+	end)
+
+	concommand.Add("-ttt2_quick_grenade", function(ply)
+		QuickGrenade(false)
+	end)
```

Comment:
I have to agree here with you because I always try to remove concommands. I'm not a fan of adding more

Reply:
I don't see the harm in adding a concommand as a matter of accessibility. Personally I made it even though the bind system exists because there are a lot of commands in TTT2 that my keyboard is too overloaded to use a key for every circumstance, which is why I use [Arctic's Radial Bind](https://steamcommunity.com/sharedfiles/filedetails/?id=2391301431). I keep the radio callouts on it, drop ammo binds, drop weapon binds. It just makes more sense to dedicate a wheel to obscure commands than crowding more keys around WASD.